src/ppx_deriving_yojson.ml

```diff
@@ -1,26 +1,26 @@
-#if OCAML_VERSION >= (4, 08, 0)
-#define Rtag(label, attrs, has_empty, args) \
-        Rtag({ txt = label }, has_empty, args)
-#elif OCAML_VERSION >= (4, 06, 0)
-#define Rtag(label, attrs, has_empty, args) \
-        Rtag({ txt = label }, attrs, has_empty, args)
-#endif
-
-#if OCAML_VERSION >= (4, 08, 0)
-#define Attribute_expr(loc_, txt_, payload) { attr_name = \
-                                                { txt = txt_; loc = loc_ }; \
-                                              attr_payload = payload; \
-                                              attr_loc = loc_ }
-#else
-#define Attribute_expr(loc_, txt_, payload) ({txt = txt_; loc = loc_}, payload)
-#endif
-
-open Longident
-open Location
-open Asttypes
-open Parsetree
+open Ppxlib
 open Ast_helper
-open Ast_convenience
+
+module Ast_builder_default_loc = struct
+  include Ppx_deriving.Ast_convenience
+
+  let gen_def_loc f x =
+    let loc = !Ast_helper.default_loc in
+    f ~loc x
+
+  let lid = gen_def_loc Ast_builder.Default.Located.lident
+  let list = gen_def_loc Ast_builder.Default.elist
+  let pstr = gen_def_loc Ast_builder.Default.pstring
+  let plist = gen_def_loc Ast_builder.Default.plist
+  let lam = gen_def_loc Ast_builder.Default.pexp_fun Nolabel None
```

Comment:
@thierry-martinez there are some missing functions in your compatibility layer. However, I don't think it matters as much as we should probably just switch to ppxlib's Ast_builder API whenever possible

Reply:
If the interest is to help plugins migrate to the ppxlib layer as easily as possible (and that sounds good to me), then I would propose to integrate those definitions in `Pppx_deriving.Ast_convenience` indeed. It looks like this new `Ast_convience` was implemented by @thierry-martinez in a more low-level way (relying on Ast_helper rather than Ast_builder), your approach may actually also suggest sensible improvements to the implementation of current Ast_convenience definitions.